catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -154,6 +151,31 @@ public MimeTypeMapper getMimeTypeMapper() {
     public void setMimeTypeMapper(MimeTypeMapper mimeTypeMapper) {
         this.mimeTypeMapper = mimeTypeMapper;
     }
+    
+    public void setRootResourceDirectories(Set<String> rootResourceDirectoryPaths) {
+        if (rootResourceDirectoryPaths == null || rootResourceDirectoryPaths.isEmpty()) {
+            LOGGER.debug("rootResourceDirectoryPaths is null or empty. Clearing rootResourceDirectories.");
+            this.rootResourceDirectories.clear();
```

Comment:
If you have to clear out the ```rootResourceDirectories``` no matter what, you could just move it out of the ```if ```statement.  Then instead of an ```if else```, you could simply do an ```if``` surrounding the ```for``` loop.